src/datasources/alarms/components/AlarmsVariableQueryEditor.tsx

```diff
@@ -24,25 +31,79 @@ export function AlarmsVariableQueryEditor({ query, onChange, datasource }: Props
 
   const onFilterChange = (filter: string) => {
     if (query.filter !== filter) {
+      query.filter = filter;
       onChange({ ...query, filter });
     }
   };
 
+  const onDescendingChange = (isDescendingChecked: boolean) => {
+    onChange({ ...query, descending: isDescendingChecked });
+  };
+
+  const validateTakeValue = (value: number) => {
+    if (isNaN(value) || value < QUERY_EDITOR_MIN_TAKE) {
+      return { message: takeErrorMessages.minErrorMsg, take: value };
+    }
+    if (value > QUERY_EDITOR_MAX_TAKE) {
+      return { message: takeErrorMessages.maxErrorMsg, take: value };
+    }
+    return { message: '', take: value };
+  };
+  
+  const onTakeChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const value = parseInt((event.target as HTMLInputElement).value, 10);
+    const { message, take } = validateTakeValue(value);
+
+    setTakeInvalidMessage(message);
+    onChange({ ...query, take });
+  };
+
   return (
     <>
-      <InlineField
-        label={labels.queryBy}
-        labelWidth={LABEL_WIDTH}
-        tooltip={tooltips.queryBy}
-      >
-        <AlarmsQueryBuilder
-          filter={query.filter}
-          onChange={(event: any) => onFilterChange(event.detail.linq)}
-          workspaces={workspaces}
-          globalVariableOptions={datasource.listAlarmsQueryHandler.globalVariableOptions()}
-        ></AlarmsQueryBuilder>
-      </InlineField>
-      <FloatingError 
+      <Stack direction="column">
+        <InlineField
+          label={labels.queryBy}
+          labelWidth={LABEL_WIDTH}
+          tooltip={tooltips.queryBy}
+        >
+          <AlarmsQueryBuilder
+            filter={filter}
+            onChange={(event: any) => onFilterChange(event.detail.linq)}
+            workspaces={workspaces}
+            globalVariableOptions={datasource.listAlarmsQueryHandler.globalVariableOptions()}
+          />
+        </InlineField>
+        <InlineField
+          label={labels.descending}
+          labelWidth={LABEL_WIDTH}
+          tooltip={tooltips.descending}
+        >
+          <InlineSwitch
+            onChange={event => onDescendingChange(event.currentTarget.checked)}
+            value={descending}
+          />
+        </InlineField>
+        <InlineField
+          label={labels.take}
+          labelWidth={LABEL_WIDTH}
+          tooltip={tooltips.take}
+          invalid={!!takeInvalidMessage}
+          error={takeInvalidMessage}
+        >
+          <AutoSizeInput
+            minWidth={LABEL_WIDTH}
+            maxWidth={LABEL_WIDTH}
+            type="number"
+            defaultValue={take}
+            onCommitChange={onTakeChange}
```

Comment:
```suggestion
            value={take}
            onChange={onTakeChange}
```

Let's avoid using deprecated props.